akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala

```diff
@@ -459,7 +459,8 @@ private[akka] class ClusterShardingGuardian extends Actor {
         val encName = URLEncoder.encode(typeName, ByteString.UTF_8)
         val cName = coordinatorSingletonManagerName(encName)
         val cPath = coordinatorPath(encName)
-        val shardRegion = context.child(encName).getOrElse {
+        val shardRegion = context.child(cName).flatMap(_ ⇒ context.child(encName)).getOrElse {
```

Comment:
I understand, also because I don't think a proxy on the same node does make much sense, are there any use cases for that ?

If you want I can implement a different naming for the proxy, otherwise do you have another suggestion on how to proceed ?